### PR TITLE
chore: DiskTree Prefix Compression

### LINKED ISSUE #565: Task: DiskTree Prefix Compression
Planning document:
- `docs/tasks/000122-disk-tree-prefix-compression.md`
- Type: task
- ID: 000122
- Title: Task: DiskTree Prefix Compression

Generated from document-first issue workflow.

### PR DESCRIPTION
**Pull request by jiangzhe**
Closes #565

## Summary by CodeRabbit

* **Refactor**
  * Reorganized B‑Tree and storage internals to centralize packing/merge logic and tighten node interfaces.
* **New Features**
  * Smarter sibling packing and merge planning with explicit fence handling and range-based node rebuilding for denser, safer blocks.
* **Bug Fixes**
  * Fixed edge cases in packing/merge and rewrite flows that could cause incorrect repacks or failed rewrites.
* **Tests**
  * Expanded tests for fence selection, space estimation, merge outcomes, rewrite/absorption, and multi-range packing.
* **Chores**
  * Removed legacy convenience scan helpers; tests and tooling now use the encoded scan API.

### CODE REVIEW

File: doradb-storage/src/index/btree/mod.rs [hunk 1]
```diff
@@ -913,33 +931,46 @@ impl<P: BufferPool> GenericBTree<P> {
...
         );
-        tmp_l.extend_slots_from::<V>(l_node, 0, l_node.count());
-        if r_node.count() > 0 {
-            tmp_l.extend_slots_from::<V>(r_node, 0, r_node.count());
-        }
-        tmp_l.update_hints();
         l_node.clone_from(&tmp_l);
         drop(tmp_l);
+        let value_size = V::ENCODED_LEN;
         p_node.delete_at(p_r_idx, value_size);
```
[the remark this answers is not in the document]
Comment (jiangzhe) [reply doradb-storage/src/index/btree/mod.rs]: Good catch.
